lib/src/models/subreddit.dart

```diff
@@ -423,14 +420,80 @@ class Subreddit extends SubredditRef with RedditBaseInitializedMixin {
   }
 }
 
-// TODO(bkonyi): implement.
-// Provides functions to interact with the special [Subreddit]'s filters.
-/*class SubredditFilters {
-  final Subreddit _subreddit;
-  SubredditFilters(this._subreddit) {
-    throw new DRAWUnimplementedError();
+/// Provides functions to interact with the special [Subreddit]'s filters.
+class SubredditFilters {
+  static final RegExp _userRegExp = new RegExp(r'{user}');
+  static final RegExp _specialRegExp = new RegExp(r'{special}');
+  final SubredditRef _subreddit;
+
+  SubredditFilters._(this._subreddit) {
+    if ((_subreddit.displayName != 'all') &&
+        (_subreddit.displayName != 'mod')) {
+      throw new DRAWArgumentError(
+          'Only special Subreddits can be filtered (r/all or r/mod)');
+    }
+  }
+
+  /// Returns a stream of all filtered subreddits.
+  Stream<SubredditRef> call() async* {
+    final user = await _subreddit.reddit.user.me();
+    final path = apiPath['subreddit_filter_list']
+        .replaceAll(_specialRegExp, _subreddit.displayName)
+        .replaceAll(_userRegExp, user.displayName);
+    final Multireddit response = await _subreddit.reddit.get(path);
+    final filteredSubs = response.data['subreddits'];
+    for (final sub in filteredSubs) {
+      yield _subreddit.reddit.subreddit(sub['name']);
+    }
+  }
+
+  /// Adds `subreddit` to the list of filtered subreddits.
+  ///
+  /// Filtered subreddits will no longer be included when requesting listings
+  /// from `/r/all`. `subreddit` can be either an instance of [String] or
+  /// [SubredditRef].
+  Future add(/* String, Subreddit */ subreddit) async {
+    var filteredSubreddit = '';
+    if (subreddit is String)
```

Comment:
Should we be standardising braces in our control flow. I know it feels nice to do this, but **if** it doesn't follow the style guide, might want to change it.

Reply:
Yes, I don't know why I did this. Good catch!